torch/lib/c10d/Utils.cpp

```diff
@@ -25,6 +27,24 @@ namespace {
 constexpr int LISTEN_QUEUE_SIZE = 2048;
 const std::string kConnectTimeoutMsg = "connect() timed out.";
 
+void checkSocketError(int socket) {
+  socklen_t errLen = sizeof(errno);
+  errno = 0;
+#ifdef _WIN32
```

Comment:
I wonder if it would worth it to create a dedicated util file to reduce the number of ifdef overall. 

E.g., does it work if we create sth like `WindowsSockUtil.h` that wraps `getsockopt`, `setsockopt`, `WSAPoll`, etc. and then we have a separate file for Unix utils. Then we just need to include different headers in this file.

Reply:
This might also help reduce the number of ifdefs in `TCPStore.cpp` and make it more readable.

Reply:
It is good idea, let me refactor in this way.